src/shared/components/lazyMobXTable/LazyMobXTable.tsx

```diff
@@ -544,58 +512,66 @@ export default class LazyMobXTable<T> extends React.Component<LazyMobXTableProps
         this.store.setProps(nextProps);
     }
 
-    buildPaginationControls(className:string | undefined, style: {[k:string]:string | number}, textBetweenButtons:string ): JSX.Element {
-        return <PaginationControls
-            className={className}
-            itemsPerPage={this.store.itemsPerPage}
-            currentPage={this.store.page}
-            onChangeItemsPerPage={this.handlers.changeItemsPerPage}
-            onPreviousPageClick={this.handlers.decPage}
-            onNextPageClick={this.handlers.incPage}
-            textBetweenButtons={textBetweenButtons}
-            previousPageDisabled={this.store.page === 0}
-            style={style}
-            nextPageDisabled={this.store.page === this.store.maxPage}
-            {...this.props.paginationProps}
-        />
+    @computed get paginationControls() {
+        return (
+            <PaginationControls
+                className="pull-left topPagination"
+                itemsPerPage={this.store.itemsPerPage}
+                currentPage={this.store.page}
+                onChangeItemsPerPage={this.handlers.changeItemsPerPage}
+                onPreviousPageClick={this.handlers.decPage}
+                onNextPageClick={this.handlers.incPage}
+                textBetweenButtons={this.store.paginationStatusText}
+                previousPageDisabled={this.store.page === 0}
+                nextPageDisabled={this.store.page === this.store.maxPage}
+                {...this.props.paginationProps}
+            />
+        );
+    }
+
+    @computed get toolbar() {
```

Comment:
we should discuss using computed on jsx like this.  the reason it worries me is that it makes the assumption that every dependency referenced is observable.  It replaces Mobx's practice of tagging components as Observers, which has the advantage that it implements shouldComponentUpdate, which can do a shallow comparison on dependencies which happen not to be observable.   Using computed to avoid React render cycles is another paradigm and we should make sure we really want to do this.